blocks/header/language-selector.js

```diff
@@ -56,66 +41,45 @@ const fetchLanguageSelectorContent = (placeholdersData, metaLangContent, langCod
       const [language, url] = pair.split(Constants.PIPE_SEPARATOR).map((part) => part.trim());
+      if (langCode === language) return;
       const liElement = li(
-        { class: `${langCode === language ? Constants.SHOW_SELECTED_CLASS : ''}` },
         a(
           { href: `${url}` },
-          getLanguageDisplayText(placeholdersData, language),
+          getLanguageDisplayText(placeholdersData, language, false),
         ),
       );
       ulElement.append(liElement);
-      // ResizeObserver for the liElement to show language code for mobile
-      const resizeObserver = createResizeObserver(placeholdersData, language);
-      resizeObserver.observe(liElement);
     });
-  } else {
-    const liElement = li(
-      { class: Constants.SHOW_SELECTED_CLASS },
-      a(
-        { href: window.location.href },
-        getLanguageDisplayText(placeholdersData, langCode),
-      ),
-    );
-    ulElement.append(liElement);
-    // ResizeObserver for the liElement to show language code for mobile
-    const resizeObserver = createResizeObserver(placeholdersData, langCode);
-    resizeObserver.observe(liElement);
   }
   return ulElement;
 };
 
 // Create and return Language Selector DOM Element
 const getLanguageSelector = (placeholdersData, lang) => {
   const metaLangContent = getMetadata(Constants.LANGUAGE_SELECTOR_META_NAME);
-  const languageMap = fetchLanguageSelectorContent(placeholdersData, metaLangContent, lang);
-
-  const languageSelectorContent = div(
-    { class: Constants.LANGUAGE_CONTENT_CLASS },
-    languageMap,
-  );
 
-  const languageToggle = div(
-    {
-      class: Constants.LANAGUAGE_TOGGLE_CLASS,
-      'aria-expanded': 'false',
-    },
-  );
-
-  const langSelector = div(
-    {
-      class: Constants.LANGUAGE_CONTAINER_CLASS,
-      onclick: (e) => toggleExpandLanguageSelector(e),
-    },
-    languageToggle,
-  );
-  langSelector.append(languageSelectorContent);
+  const languageToggle = div({ 'aria-expanded': 'false' });
+  const langSelector = div({ class: Constants.LANGUAGE_CONTAINER_CLASS }, languageToggle);
 
+  // Show only Current Language when no meta content authored
+  if (!metaLangContent) {
+    updateLanguageTextContent(languageToggle, placeholdersData, lang);
+    languageToggle.classList.add(Constants.LANGUAGE_TEXT_CLASS);
+  } else {
+    const languageMap = fetchLanguageSelectorContent(placeholdersData, metaLangContent, lang);
+    const languageSelectorContent = div(
+      { class: Constants.LANGUAGE_CONTENT_CLASS },
+      languageMap,
+    );
+    languageToggle.classList.add(Constants.LANAGUAGE_TOGGLE_CLASS);
+    langSelector.addEventListener('click', toggleExpandLanguageSelector);
+    langSelector.append(languageSelectorContent);
+  }
   // Show Language code in Mobile
   const resizeObserver = new ResizeObserver(() => {
     updateLanguageTextContent(languageToggle, placeholdersData, lang);
   });
-
-  resizeObserver.observe(langSelector);
+  resizeObserver.observe(document.body);
```

Comment:
See if you can observe the `header` element instead. Also, keep it inside a check so you don't get console errors. If that element is not available in DOM, in your else condition, you can observe `document` so that you don't get console errors.

Reply:
Done